modules/gapi/include/opencv2/gapi/s11n.hpp

```diff
@@ -142,7 +154,105 @@ namespace I {
     GAPI_EXPORTS I::OStream& operator<< (I::OStream& os, const cv::gapi::wip::draw::Line &l);
     GAPI_EXPORTS I::IStream& operator>> (I::IStream& is,       cv::gapi::wip::draw::Line &l);
 
-    // Generic stl structures
+    // G-API types /////////////////////////////////////////////////////////////////
+
+    GAPI_EXPORTS I::OStream& operator<< (I::OStream& os, cv::util::monostate  );
+    GAPI_EXPORTS I::IStream& operator>> (I::IStream& is, cv::util::monostate &);
+
+    GAPI_EXPORTS I::OStream& operator<< (I::OStream& os, cv::GShape  shape);
+    GAPI_EXPORTS I::IStream& operator>> (I::IStream& is, cv::GShape &shape);
+
+    GAPI_EXPORTS I::OStream& operator<< (I::OStream& os, cv::detail::ArgKind  k);
+    GAPI_EXPORTS I::IStream& operator>> (I::IStream& is, cv::detail::ArgKind &k);
+
+    GAPI_EXPORTS I::OStream& operator<< (I::OStream& os, cv::detail::OpaqueKind  k);
+    GAPI_EXPORTS I::IStream& operator>> (I::IStream& is, cv::detail::OpaqueKind &k);
+
+    GAPI_EXPORTS I::OStream& operator<< (I::OStream& os, const cv::GArg &arg);
+    GAPI_EXPORTS I::IStream& operator>> (I::IStream& is,       cv::GArg &arg);
+
+
+    GAPI_EXPORTS I::OStream& operator<< (I::OStream& os, const cv::GKernel &k);
+    GAPI_EXPORTS I::IStream& operator>> (I::IStream& is,       cv::GKernel &k);
```

Comment:
This shouldn't be here too, I believe. Please keep it in the internal header.

Reply:
Removed